vhost/src/vhost_user/backend_req.rs

```diff
@@ -131,11 +143,17 @@ impl Backend {
 impl VhostUserFrontendReqHandler for Backend {
     /// Forward vhost-user shared-object add request to the frontend.
     fn shared_object_add(&self, uuid: &VhostUserSharedMsg) -> HandlerResult<u64> {
+        if !self.node().shared_object_negotiated {
+            return Ok(1);
```

Comment:
Should we return an error?

Reply:
I hesitated here.

So the spec states: `the front-end must respond with zero when operation is successfully completed, or non-zero otherwise.`, so returning `Ok(1)` is interpreted as an error by the device.

On the other hand, this is not the frontend setting the return value. So maybe it would be better to return an error. But if I do, which one? Should I create one `ErrorType`?

Reply:
IIUC these are the APIs called from the backend, so before we even send the message to the frontend. So I think it's an implementation detail of ours.

What about the `Error::InactiveFeature(VhostUserVirtioFeatures)` already defined in https://github.com/rust-vmm/vhost/blob/043d0d8fa2037da47f4a621da6c13d5f379b7c69/vhost/src/vhost_user/mod.rs#L64

Reply:
Yes, this occurs before sending the message to the frontend.

Uhm, it needs to be an IoError... What about `return Err(std::io::Error::from(ErrorKind::InvalidInput));`?

Reply:
Ah, right, or `ErrorKind::Unsupported`. Is up to you ;-)

Reply:
Pushed with `ErrorKind::Other` in the end. `ErrorKind::Unsupported` would be a possibility but sounds like it is not supported at all to send those messages, so I thought that better to be explicit and let the device know the exact reason.